src/ngraph/op/extractimagepatches.cpp

```diff
@@ -0,0 +1,157 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/extractimagepatches.hpp"
+#include "ngraph/attribute_visitor.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+// ExtractImagePatches v3
+
+constexpr NodeTypeInfo op::v3::ExtractImagePatches::type_info;
+
+op::v3::ExtractImagePatches::ExtractImagePatches(const Output<Node>& image,
+                                                 const Shape& sizes,
+                                                 const Strides& strides,
+                                                 const Shape& rates,
+                                                 const PadType& padding)
+    : Op({image})
+    , m_patch_sizes(sizes)
+    , m_patch_movement_strides(strides)
+    , m_patch_selection_rates(rates)
+    , m_padding(padding)
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v3::ExtractImagePatches::validate_and_infer_types()
+{
+    const PartialShape input_Pshape = get_input_partial_shape(0);
+
+    NODE_VALIDATION_CHECK(this,
+                          get_input_element_type(0).is_integral_number(),
+                          "input tensor must be an integral number.");
+    NODE_VALIDATION_CHECK(this, input_Pshape.rank() == 4, "input tensor must be 4D tensor.");
+
+    NODE_VALIDATION_CHECK(this,
+                          m_patch_sizes.size() == 2,
+                          "Attribute sizes should be in [size_rows, size_cols] format.");
+
+    NODE_VALIDATION_CHECK(this,
+                          m_patch_movement_strides.size() == 2,
+                          "Attribute strides should be in [stride_rows, stride_cols] format.");
+
+    NODE_VALIDATION_CHECK(this,
+                          m_patch_movement_strides[0] > 0 && m_patch_movement_strides[1] > 0,
+                          "Attribute strides should be strictly greater than zeros in values.");
+
+    NODE_VALIDATION_CHECK(this,
+                          m_patch_selection_rates.size() == 2,
+                          "Attribute rates should be in [rate_rows, rate_cols] format.");
+
+    NODE_VALIDATION_CHECK(this,
+                          m_patch_selection_rates[0] > 0 && m_patch_selection_rates[1] > 0,
+                          "Attribute rates should be strictly greater than zeros in values.");
+
+    NODE_VALIDATION_CHECK(
+        this,
+        m_padding == PadType::VALID || m_padding == PadType::SAME_LOWER ||
+            m_padding == PadType::SAME_UPPER,
+        "Attribute padding should be in either valid or same_lower or same_upper.");
+
+    if (input_Pshape.is_dynamic())
+    {
+        set_input_is_relevant_to_shape(0);
```

Comment:
If only the batch dimension is dynamic we still can calculate the 3 dimensions out of 4. So please, change the condition in line 76 so the "else" branch will work for case when the only batch dimension is dynamic. Also add a unit test for this case.

Reply:
fixed according to your suggestion. Thanks.